### ISSUE TITLE
Port for macOS

### ISSUE THREAD

**Issue by BH-Playz** [Metabolix/HackBGRT#150]
Hey, I recently switched back to macOS and want to keep my custom cool logo, Could you please add a macOS.dmg file? Maybe an ARM dmg?

**Comment by sylveon** [comment]
This is an UEFI application, it should be OS independent. Unknown if macOS respects the BGRT however.

**Comment by Metabolix** [comment]
I don't have a Mac. Maybe you can fix that?

Otherwise, try to copy the files manually (maybe not necessary if you didn't wipe it when switching OS) and add the boot entry manually woth some tool such as [bootoption](https://github.com/bootoption/bootoption).

**Comment by womeimingzi11** [comment]
From what I can tell, Macs do not use the UEFI to boot computer. I am afraid it is not easy to port HGRT things.

**Comment by sylveon** [comment]
The x86-based Macs do. No idea if the ARM64 ones do however.

**Comment by womeimingzi11** [comment]
Yes, you. are. right. The x86 Macs just work like a general PC. And this
the reason why Hackintosh works.

But the aarch64 based Macs do not have UEFI.

Charles Milette ***@***.***> 于 2024年3月24日周日 下午3:46写道：

**Comment by Yarpopcat08** [comment]
Don't x86 macs use their own boot logo instead of the one in BGRT?

**Comment by Metabolix** [comment]
> Don't x86 macs use their own boot logo instead of the one in BGRT?

Could someone check this out on a Mac which dual-boots to either Windows or Linux (with _plymouth_ splash screen)?

**Comment by Metabolix** [comment]
From the source code of [BootXChanger](https://github.com/zydeco/bootxchanger) (later borrowed into [DarkBoot](https://github.com/w0lfschild/DarkBoot)), looks like BGRT is not involved and it's simply a matter of adding a PNG path into `/Library/Preferences/SystemConfiguration/com.apple.Boot.plist`.

``` xml
<key>Boot Logo</key>
<string>\System\Library\CoreServices\BootLogo.png</string>
```

I will close this issue for now.
